app/models/auditing/UserHasPenaltyAuditModel.scala

```diff
@@ -136,8 +137,12 @@ case class UserHasPenaltyAuditModel(
       lppDetails <- lpp.details
       optSeqTimeToPay <- lppDetails.find(_.metadata.timeToPay.isDefined).map(_.metadata.timeToPay.get)
       optActiveTimeToPay <- optSeqTimeToPay.find(
-        penalty => penalty.TTPEndDate.exists(DateHelper.isDateAfterOrEqual(_, dateNow)) &&
-          penalty.TTPStartDate.exists(DateHelper.isDateBeforeOrEqual(_, dateNow)))
+        penalty => (penalty.TTPEndDate, penalty.TTPStartDate) match {
+          case (Some(endDate), Some(startDate)) => startDate < dateNow && dateNow < endDate
+          case (None, Some(startDate)) => startDate < dateNow
```

Comment:
I'm not a fan of treating dates as something that can be compared via '>', '<' etc., but that's personal taste.

However, I think these operators should be 'x than _or equal to_' i.e.,
```suggestion
          case (Some(endDate), Some(startDate)) => startDate <= dateNow && dateNow <= endDate
          case (None, Some(startDate)) => startDate <= dateNow
```

Because it currently returns false if today's date is equal to the TTP start/end date

Reply:
Where x is 'less than' or 'more than'

Reply:
Switched back to DateHelper for clarity ✅